app/components/Authentication/Github/Github.jsx

```diff
@@ -1,248 +1,513 @@
-                this.setState({ errorMessage: `${err} - URI: ${decodeURI(uri)}` });
-            });
     }
 
-    teamNameChanged(e, v) {
-        if (v === '') {
-            _.forEach(this.state.policies, (policy) => {
-                policy.checked = false;
-            });
-
-            this.setState({
-                policies: this.state.policies
-            });
+    componentDidMount() {
+        if (this.props.params.splat) {
+            this.setState({ selectedItemId: this.props.params.splat });
         } else {
-            this.state.teamName = v;
-            this.processTeamNameDebounced();
+            this.listGithubTeams();
+            this.listGithubUsers();
         }
+        this.getOrgConfig();
+    }
 
+    componentDidUpdate(prevProps, prevState) {
+        if (this.state.selectedItemId != prevState.selectedItemId) {
+            this.listGithubTeams();
+            this.listGithubUsers();
+            if (this.state.selectedItemId) {
+                this.displayItem();
+            }
+        }
     }
 
-    processTeamName() {
-        this.getTeamPolicy(this.state.teamName);
-        this.setState({
-            teamName: this.state.teamName,
-            submitBtnDisabled: !(this.state.checkedPolicies && this.state.organization && this.state.teamName),
-            submitBtnColor: (this.state.checkedPolicies && this.state.organization && this.state.teamName) ? green500 : 'lightgrey'
-        });
+    componentWillReceiveProps(nextProps) {
+        if (!_.isEqual(this.props.params.namespace, nextProps.params.namespace)) {
+            // Reset
+            this.setState({
+                baseUrl: `/auth/github/${nextProps.params.namespace}/`,
+                baseVaultPath: `auth/${nextProps.params.namespace}`,
+                users: [],
+                teams: [],
+                selectedItemId: '',
+                newConfig: this.backendConfigSchema,
+                config: this.backendConfigSchema,
+                selectedTab: "team"
+            }, () => {
+                this.listGithubTeams();
+                this.listGithubUsers();
+                this.getOrgConfig();
+            });
+        }
     }
 
-    policyChecked(policyName, e, isChecked) {
-        let policies = this.state.policies;
+    render() {
+        let renderListItems = () => {
+            let items = this.state.selectedTab === "team" ? this.state.teams : this.state.users;
+            return _.map(items, (item) => {
+                let avatar = (<Avatar icon={<ActionAccountBox />} />);
+                let action = (
+                    <IconButton
+                        tooltip="Delete"
+                        onTouchTap={() => this.setState({ deleteUserPath: `${this.state.baseVaultPath}/map/${this.state.selectedTab}s/${item}` })}
+                    >
+                        {window.localStorage.getItem("showDeleteModal") === 'false' ? <ActionDeleteForever color={red500} /> : <ActionDelete color={red500} />}
```

Comment:
The value we're comparting to is of type string, so we can't use a falsy in this case.